### PR TITLE
use `Buffer.from` rather than `new Buffer []`

### PR DESCRIPTION
**Pull request by maxtaco**
- get all tests passing, and increase the in-browser test set

### CODE REVIEW

**Comment by maxtaco** [comment]
This is because `new Buffer [1,2,3]` is [deprecated in node 6+](https://nodejs.org/dist/latest-v6.x/docs/api/buffer.html#buffer_new_buffer_array).

**Comment by zapu** [comment]
image

In Firefox :(

**Comment by zapu** [comment]
It looks like there is a number of `new Buffer` still there:

e.g.
https://github.com/keybase/kbpgp/blob/44a0e554f01143f4c788f021b526916cdfdb8045/src/keybase/hilev.iced

Documentation seems to say that all uses of `new Buffer` are deprecated https://nodejs.org/api/buffer.html

Do we leave the "non-obvious" ones for now?

**Comment by maxtaco** [comment]
I don't think Firefox should ever see this, since it's being polyfilled by Browserify, no? Can you try `make test-browser` in Firefox?

**Comment by zapu** [comment]
This was from `make test-browser`, sorry for not mentioning it.

**Comment by maxtaco** [comment]
Weird, I can't repro. Works fine for me on Firefox. Btw, in Chrome debugging, `Buffer.from` doesn't work either

Those remnant `new Buffer` are not using an array as an argument. They are either using an integer (for size of buffer to allocate) or are using string initialization.  I think both of those modes of operation are not deprecated.

**Comment by maxtaco** [comment]
(I'm running 46.0.1)

**Comment by maxtaco** [comment]
@zapu can you post a stack trace from your Firefox failure?

**Comment by zapu** [comment]
I had to leave for a while, sorry. It was failing in curve oid definitions IIRC. I'll try it some more when I get back, maybe it didn't buils correctly. 

Dnia 11.04.2017 o godz. 15:23 Maxwell Krohn <notifications@github.com> napisał(a):

**Comment by zapu** [comment]
Looks like it was old browserify version, sorry for that.

**Comment by maxtaco** [comment]
no worries

**Comment by zapu** [comment]
otherwise looks good, I'm only worried that they are deprecating all of the constructors so we might have to do more refactoring later.

**Comment by maxtaco** [comment]
Whoops, you're right, I didn't see that at first.  I should go back and fix the rest.

**Comment by maxtaco** [comment]
i'll fix it!

File: src/openpgp/ocfb.iced [hunk 1]
```diff
@@ -279,9 +279,9 @@ exports.Decryptor = Decryptor
 
 {rng} = require 'crypto'
 test = () ->
-  plaintext = new Buffer("a man a plan a canal panama. and you know the rest")
+  plaintext = Buffer.from("a man a plan a canal panama. and you know the rest", 'utf8')
```
Comment (zapu) [inline src/openpgp/ocfb.iced]: Looks like default encoding is `utf8` for both `new Buffer` and `Buffer.from`, but good call on passing it explicitly.

**Comment by maxtaco** [comment]
See #181

**Pull request closed without merging**